src/test/java/com/datadoghq/datadog_lambda_java/DDLambdaInstantiationTest.java

```diff
@@ -0,0 +1,112 @@
+import com.amazonaws.services.lambda.runtime.events.APIGatewayV2ProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.KinesisEvent;
+import com.amazonaws.services.lambda.runtime.events.SQSEvent;
+import com.datadoghq.datadog_lambda_java.EnhancedMetricTest.MockContext;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.EnvironmentVariables;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class DDLambdaInstantiationTest {
+
+    private static final String API_GATEWAY_PROXY_TYPE = "com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent";
+    private static final String API_GATEWAY_V2_PROXY_TYPE = "com.amazonaws.services.lambda.runtime.events.APIGatewayV2ProxyRequestEvent";
+    private static final String SQS_EVENT_TYPE = "com.amazonaws.services.lambda.runtime.events.SQSEvent";
+    private static final String KINESIS_EVENT_TYPE = "com.amazonaws.services.lambda.runtime.events.KinesisEvent";
+    private static final String NO_EVENT = "null";
+    private final Object event;
+    private String eventType;
+    private final Context context;
+    private final Boolean tracingEnabled;
+
+    @Rule
+    public final EnvironmentVariables environmentVariables = new EnvironmentVariables();
+
+    public DDLambdaInstantiationTest(Object event, Context mockContext, Boolean tracingEnabled) {
+        this.event = event;
+        this.context = mockContext;
+        this.tracingEnabled = tracingEnabled;
+    }
+
+    @Parameters
+    public static Collection<Object[]> data() {
+        APIGatewayProxyRequestEvent dummyAPIGateway = new APIGatewayProxyRequestEvent();
+        dummyAPIGateway.setHeaders(new HashMap<String, String>());
+        APIGatewayV2ProxyRequestEvent dummyAPIGatewayV2 = new APIGatewayV2ProxyRequestEvent();
+        dummyAPIGatewayV2.setHeaders(new HashMap<String, String>());
+        return Arrays.asList(new Object[][]{
+                //Event, Context, DD_TRACE_ENABLED value
+                { dummyAPIGateway, new MockContext(), true },
+                { dummyAPIGateway, new MockContext(), false },
+                { dummyAPIGatewayV2, new MockContext(), true },
+                { dummyAPIGatewayV2, new MockContext(), false },
+                { new SQSEvent(), new MockContext(), true },
+                { new SQSEvent(), new MockContext(), false },
+                { new KinesisEvent(), new MockContext(), true },
+                { new KinesisEvent(), new MockContext(), false },
+                { null, new MockContext(), true },
+                { null, new MockContext(), false },
+                { null, null, true },
+                { null, null, false },
+        });
+    }
```

Comment:
This is a [Parameterized test](https://github.com/junit-team/junit4/wiki/Parameterized-tests). It lets me create a bunch of similar tests (in this case 12), with different inputs.